tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-dashboard.html

```diff
@@ -858,11 +858,11 @@
           // TODO(cais): If the TDP backend is not activated, display message
           // informing the user of that.
           this.$.initialDialog.openDialog(response.host, response.port);
-        }).catch(error => {
-          this.$.initialDialog.openDisabledDialog();
         }).then(() => {
           // Initiate long-polling.
           this.long_poll();
+        }).catch(error => {
```

Comment:
Honestly, this change confused me and I still think it is confusing. Please correct me but it looks like this change is making use of the fact that `openDialog` now throws an exception when either host or port is empty. The flow of thought jumps needlessly across modules and I am not sure if this is good.

Also, we should try to have a principle of having business logic higher up in the component. The InitialDialog (I think the name should be changed too or there should be a separate class that is for disabled state) should be only about presentation of the error state and whether something is valid/invalid should live in this component. If you care more about the separation of concern, I think, then, the InitialDialog should be able to detect the error case and show the disabled dialog itself (which I can see making things more complex).